os/arch/arm/src/armv8-m/up_assert.c

```diff
@@ -275,52 +276,78 @@ static void up_dumpstate(void)
 
 	ustackbase = (uint32_t)rtcb->adj_stack_ptr;
 	ustacksize = (uint32_t)rtcb->adj_stack_size;
+
 	lldbg("sp:     %08x\n", sp);
 
+	/* Show the interrupt number, handler and stack information
+	 * if we are in an interrupt handler (g_irq_nums[1] != 0)
+	 */
+
+	int assertInIrq = g_irq_nums[1];
+
+	if (assertInIrq) {
+#ifdef CONFIG_ARCH_NESTED_IRQ_STACK_SIZE
+#ifdef CONFIG_STACK_COLORATION
+		if (up_check_nestirqstack()) {
```

Comment:
`up_check_nestirqstack` does not indicate that the crash happened in nested irq. I think , you will have to maintain a global array of 3 and store previous 3 irqs. If the third one is non zero, then it indicates that crash happened in nested irq. Else, if 2nd position is non zero, it indicates crash in normal irq. Else it indicates crash in user thread.

Reply:
Updated the code. Thank you.